### PR TITLE
Automatic Deploy

### PR DESCRIPTION
**Pull request by worldbeater**
**What kind of change does this PR introduce?**

This PR implements automatic deploy of the artifacts built by Azure Pipelines to a dedicated server via SSH. With this PR, the general deployment process consists of the following steps:
1. Run `ava.service.sh` on a clean Ubuntu machine. This installs .NET Core 3.1 and configures automatic restarts on failures.
2. Add a new service connection to the host machine via Azure DevOps (`Project Settings -> Pipelines -> Service Connections -> New service connection -> SSH`), service connection name should be `production-server`.
2. Push something to the main `master` branch and the app will auto-reload with new content after a successful CI build.

I've set up a sample CI for my `avaloniaui.net` repository fork so one could view an example of a successful job https://worldbeater.visualstudio.com/Camelotia/_build/results?buildId=567&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9

### CODE REVIEW

**Comment by worldbeater** [comment]
Now hitting an unrelated but blocking thing — a weird exception from `YamlDotNet` is thrown when one tries to open `docs`. Easily reproducible — just clone the repo (any branch), `dotnet run` the project, and then click `Documentation`. Still investigating what's causing this. 
image

File: azure-pipelines.yml [hunk 1]
```diff
@@ -25,3 +25,41 @@ steps:
...
+    cleanTargetFolder: true
+    overwrite: true
+  condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+- task: SSH@0
+  inputs:
+    displayName: 'Start Host App'
+    sshEndpoint: 'production-server'
+    runOptions: 'commands'
+    commands: |
+     sudo systemctl stop ava.service > /dev/null 2>&1
```
Comment (kekekeks) [inline azure-pipelines.yml:56]: Definitely not allowing root access for CI server user

Comment (worldbeater) [reply azure-pipelines.yml:56]: Then the whole idea with using `systemd` won't work I guess, so not much sense in this PR anymore. Probably we could consider using docker, but then there isn't much sense in Azure Pipelines as we can check out, build and run the image on a dedicated server without the need for this `ssh` dance with an intermediate machine.

Comment (worldbeater) [reply azure-pipelines.yml:56]: Would highly appreciate any idea about what we should use instead of `systemd` here.

Comment (kekekeks) [reply azure-pipelines.yml:56]: The usual way to trigger an update is to have a predefined script and then allow its execution with NOPASSWD from /etc/sudoers. Said script should be hardcoded as the only allowed command for CI ssh key in ~/.ssh/authorized_keys

Comment (worldbeater) [reply azure-pipelines.yml:56]: Ah I see, so this means that we can still use systemd here. Thanks for the guidance, going to implement the suggested solution!

**Comment by worldbeater** [comment]
Closing the PR intending not to disturb the current CI.

**Pull request closed without merging**